homeassistant/components/radiotherm/config_flow.py

```diff
@@ -0,0 +1,171 @@
+        )
+
+    async def async_step_import(self, import_info: dict[str, Any]) -> FlowResult:
+        """Import from yaml."""
+        host = import_info[CONF_HOST]
+        self._async_abort_entries_match({CONF_HOST: host})
+        _LOGGER.debug("Importing entry for host: %s", host)
+        try:
+            init_data = await validate_connection(self.hass, host)
+        except CannotConnect as ex:
+            _LOGGER.debug("Importing failed for %s", host, exc_info=ex)
+            return self.async_abort(reason="cannot_connect")
+        await self.async_set_unique_id(init_data.mac, raise_on_progress=False)
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: host}, reload_on_update=False
+        )
+        return self.async_create_entry(
+            title=init_data.name,
+            data={CONF_HOST: import_info[CONF_HOST]},
+            options={CONF_HOLD_TEMP: import_info[CONF_HOLD_TEMP]},
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                init_data = await validate_connection(self.hass, user_input[CONF_HOST])
+            except CannotConnect:
+                errors[CONF_HOST] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(init_data.mac, raise_on_progress=False)
+                self._abort_if_unique_id_configured(
+                    updates={CONF_HOST: user_input[CONF_HOST]},
+                    reload_on_update=False,
+                )
+                return self.async_create_entry(
+                    title=init_data.name,
+                    data=user_input,
+                    options={CONF_HOLD_TEMP: False},
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for radiotherm."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
```

Comment:
This really should be a switch entity instead but I didn't want to add a new platform since this PR is already large.  I'll do that in another turn

Reply:
ok. Would be great if we can do it before the next release so it's not a breaking change.

Reply:
Marking this as a draft so it doesn't get merged prematurely.  Once I have the switch platform ready, I'll merge this and open the other PR.

Reply:
I've got the new PR prepared so I'm going to merge now and fix the conflict